src/utils/thread_helper.cpp

```diff
@@ -1360,8 +1392,48 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
         }
         // End of Nat64Mapping section.
 #endif // OTBR_ENABLE_NAT64
+#if OTBR_ENABLE_DHCP6_PD
+        // Start of Dhcp6PdState section.
+        {
+            wpanBorderRouter->set_dhcp6_pd_state(
+                Dhcp6PdStateFromOtDhcp6PdState(otBorderRoutingDhcp6PdGetState(mInstance)));
+        }
+        // End of Dhcp6PdState section.
 
-        // End of WpanBorderRouter section.
+        // Start of Hashed PD prefix
+        {
+            otBorderRoutingPrefixTableEntry aPrefixInfo;
+            Sha256::Hash                    hash;
+            Sha256                          sha256;
+
+            otBorderRoutingGetPdOmrPrefix(mInstance, &aPrefixInfo);
+            uint8_t pdPrefixShaInput[OT_IP6_ADDRESS_SIZE + kNat64PdCommonHashSaltLength];
+            memcpy(pdPrefixShaInput, aPrefixInfo.mPrefix.mPrefix.mFields.m8,
```

Comment:
Suggestion to simplify this code:
- We dont need to `memcpy` to construct the whole `pdPrefixShaInput` buffer by appending the prefix first followed by the slat.
- The `Sha256::Update()` method can be called multiple times.
- We can read the prefix, feed it directly in `Sha256::Update()`
- Then call `Sha256::Update()` again to feed the salt bytes.